src/main/java/org/mule/extension/runtime/ExpirationPolicy.java

```diff
@@ -0,0 +1,35 @@
+package org.mule.extension.runtime;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A policy around when should a given instances be considered elegible for expiration.
```

Comment:
What do you mean with "elegible for expiration"? Is there another component that can also decide wether or not to expire something that was "elegible for expiration"?